### PR TITLE
Use yarn 3 and update to latest sdk

### PR DESCRIPTION
**Pull request by yorhodes**

### CODE REVIEW

File: src/deploy/deploy.ts [hunk 1]
```diff
@@ -1,38 +1,83 @@
...
+
+    const abacusConnectionManager =
+      await this.deployConnectionManagerIfNotConfigured(network);
+
+    const upgradeBeaconController = await this.deployContract(
+      network,
+      'UpgradeBeaconController',
+      new UpgradeBeaconController__factory(signer),
+      [],
+    );
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply src/deploy/deploy.ts:47]: https://github.com/abacus-network/abacus-monorepo/issues/373

File: src/deploy/deploy.ts [hunk 2]
```diff
@@ -1,38 +1,83 @@
...
     };
   }
 
-  mustGetRouter(domain: number): PingPong {
+  mustGetRouter(network: Networks, addresses: PingPongAddresses) {
     return PingPong__factory.connect(
-      this.mustGetAddresses(domain).router,
-      this.mustGetSigner(domain),
+      addresses.router.proxy,
+      this.multiProvider.getDomainConnection(network).signer!,
```
Comment (yorhodes) [inline src/deploy/deploy.ts:80]: this seems like something that can happen in `AbacusRouterDeployer`

File: contracts/PingPong.sol [hunk 3]
```diff
@@ -30,8 +30,13 @@ contract PingPong is Router {
...
 
     // ============ Constructor ============
-    constructor(address _connectionManager) {
-        __Router_initialize(_connectionManager);
+
+    constructor() {}
+
+    // ============ Initializer ============
+
+    function initialize(address _abacusConnectionManager) public initializer {
```
Comment (asaj) [inline contracts/PingPong.sol:38]: External?

File: src/deploy/deploy.ts [hunk 4]
```diff
@@ -1,38 +1,83 @@
...
+  PingPongConfig<Networks>,
+  PingPongAddresses
 > {
+  constructor(
+    multiProvider: MultiProvider<Networks>,
+    config: PingPongConfig<Networks>,
+    core?: AbacusCore<Networks>,
+  ) {
+    const networks = multiProvider.networks();
+    const crossConfigMap = Object.fromEntries(
```
Comment (asaj) [inline src/deploy/deploy.ts:26]: Why is this necessary?

File: src/deploy/deploy.ts [hunk 5]
```diff
@@ -1,38 +1,83 @@
...
-    const xAppConnectionManager =
-      await this.deployConnectionManagerIfNotConfigured(domain, config);
-
-    const signer = this.mustGetSigner(domain);
-    const router = await this.deployContract(
-      domain,
+    network: Networks,
+    config: PingPongConfig<Networks>,
+  ): Promise<PingPongAddresses> {
+    const dc = this.multiProvider.getDomainConnection(network);
```
Comment (asaj) [inline src/deploy/deploy.ts:36]: Is there not a `multiProvider.getSigner(network)` function that we could call instead?

File: src/deploy/scripts/check.ts [hunk 6]
```diff
@@ -1,24 +1,26 @@
...
 import { PingPongChecker } from '../check';
-import { getEnvironmentConfig } from './utils';
+import { configs } from '../networks';
 
 async function check() {
-  const environment = await utils.getEnvironment();
-  const pingPong = new PingPongApp(addresses);
-  const config = await getEnvironmentConfig(environment as any);
-  await utils.registerEnvironment(pingPong, config);
+  const transactionConfigs = {
```
Comment (asaj) [inline src/deploy/scripts/check.ts:9]: Should this be exported by `networks`? Otherwise won't this object need to be created in a bunch of different places?

File: src/deploy/scripts/deploy.ts [hunk 7]
```diff
@@ -1,19 +1,23 @@
...
-  const environment = await utils.getEnvironment();
-  const core = new AbacusCore(coreAddresses);
-  const deployer = new PingPongDeployer(core);
-  const config = await getEnvironmentConfig(environment as any);
-  await utils.registerEnvironment(core, config);
-  await utils.registerEnvironment(deployer, config);
-  await deployer.deploy(utils.getRouterConfig(core) as any); // TODO: fix types
-
-  deployer.writeOutput(path.join('./src/deploy/output/', environment));
+  const transactionConfigs = {
```
Comment (asaj) [inline src/deploy/scripts/deploy.ts:9]: See previous comment

File: src/deploy/scripts/check.ts (hunk 6 above)
Comment (nambrot) [reply src/deploy/scripts/check.ts:9]: +1

File: src/deploy/scripts/deploy.ts (hunk 7 above)
Comment (nambrot) [reply src/deploy/scripts/deploy.ts:9]: +1

File: src/deploy/check.ts [hunk 8]
```diff
@@ -1,15 +1,24 @@
...
-    return this.app.mustGetContracts(domain).router;
+  async checkDomainAddresses(
+    network: Networks,
+    owner: types.Address,
+  ): Promise<void> {
+    await super.checkDomain(network, owner);
+  }
+
+  mustGetRouter(network: Networks) {
+    return this.app.getContracts(network).router;
```
[the remark this answers is not in the document]
Comment (nambrot) [reply src/deploy/check.ts:22]: Doesn't look like it

File: src/deploy/deploy.ts [hunk 9]
```diff
@@ -1,38 +1,83 @@
...
+    multiProvider: MultiProvider<Networks>,
+    config: PingPongConfig<Networks>,
+    core?: AbacusCore<Networks>,
+  ) {
+    const networks = multiProvider.networks();
+    const crossConfigMap = Object.fromEntries(
+      networks.map((network) => [network, config]),
+    ) as ChainMap<Networks, PingPongConfig<Networks>>;
+    super(multiProvider, crossConfigMap, core);
+  }
```
[the remark this answers is not in the document]
Comment (nambrot) [reply src/deploy/deploy.ts:30]: Ideally shouldn't be or shouldn't right now? Cause I'm pretty sure I needed this to make the compiler happy

File: src/deploy/deploy.ts (hunk 4 above)
Comment (nambrot) [reply src/deploy/deploy.ts:26]: See below

File: src/deploy/deploy.ts (hunk 2 above)
Comment (nambrot) [reply src/deploy/deploy.ts:80]: +1

File: src/sdk/app.ts [hunk 10]
```diff
@@ -1,23 +1,37 @@
...
   async pingRemote(
-    from: NameOrDomain,
-    to: NameOrDomain,
+    from: Networks,
+    to: Networks,
   ): Promise<ethers.ContractReceipt> {
-    const router = this.mustGetContracts(from).router;
-    const toDomain = this.resolveDomain(to);
+    const router = this.getContracts(from).router;
+    const toDomain = ChainNameToDomainId[to];
```
[the remark this answers is not in the document]
Comment (nambrot) [reply src/sdk/app.ts:34]: Like we discussed, I don't think we should abstract too much. The goal isn't for the developer to write the least amount of code, but to be most successful in writing an app. I'm afraid we are doing too much at the expense of the developer being able to understand what is happening

File: src/sdk/types.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import { RouterConfig } from '@abacus-network/deploy';
+import { ChainName } from '@abacus-network/sdk';
+
+export type PingPongConfigAddresses = {
```
[the remark this answers is not in the document]
Comment (nambrot) [reply src/sdk/types.ts:4]: Remnant from when YoConfig was assuming that addresses should be configured (like abacus connectin managers)

File: src/deploy/deploy.ts (hunk 5 above)
Comment (nambrot) [reply src/deploy/deploy.ts:35]: https://github.com/abacus-network/abacus-monorepo/issues/394#issuecomment-1119195822

**Review by yorhodes — COMMENTED** [review]
these comments might be stale
cant approve

**Review by nambrot — APPROVED** [review]

File: src/sdk/environments/test.ts [hunk 12]
```diff
@@ -1,18 +1,29 @@
...
-    router: '0xc0F115A19107322cFBf1cDBC7ea011C19EbDB4F8',
-    xAppConnectionManager: '0xDC11f7E700A4c898AE5CAddB1082cFfa76512aDD',
+  test3: {
+    router: {
+      proxy: '0x0DCd1Bf9A1b36cE34237eEaFef220932846BCD82',
+      implementation: '0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e',
+      beacon: '0xA51c1fc2f0D1a1b8494Ed1FE312d7C3a78Ed91C0',
+    },
+    upgradeBeaconController: '0x610178dA211FEF7D417bC0e6FeD39F05609AD788',
+    abacusConnectionManager: '0x70e0bA845a1A0F2DA3359C97E0285013525FFC49',
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply src/sdk/environments/test.ts]: maybe we should have an `init` like the `hardhat` cli does

[the remark this answers is not in the document]
Comment (nambrot) [reply src/sdk/environments/test.ts]: This internal circular dependency is quite awkward
